### PR TITLE
Persist alignment when transforming a gallery to an image and vice-versa

### LINKED ISSUE #11374: When changing a gallery block to multiple images, let the image blocks inherit the gallery alignment
**Is your feature request related to a problem? Please describe.**
My blog is "old school" in that it just posts image after image, each with a caption.  In the old editor I upload the images into the Media Library, give them captions, submit them all.  Now I have to start a gallery block, upload the images, give them captions, then insert into the page.  Then I need to change the gallery block into images, except the images all appear left aligned (breaking the style sheet?) and I have to individually align each image block to be the centered.

**Describe the solution you'd like**
I can give the gallery an alignment, then explode it into image blocks, and all the image blocks inherit that alignment.

**Describe alternatives you've considered**
Let me set the image blocks to be center aligned by default.

### PR DESCRIPTION
**Pull request by andreilupu**
A fresh rebase for #11651 
Fixes: https://github.com/WordPress/gutenberg/issues/11374

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/gallery/index.js [hunk 1]
```diff
@@ -100,13 +100,28 @@ export const settings = {
 				blocks: [ 'core/image' ],
 				transform: ( attributes ) => {
 					const validImages = filter( attributes, ( { id, url } ) => id && url );
+					// init the align attribute from the first item which may be either the place holder or an image.
```
Comment (jorgefilipecosta) [inline packages/block-library/src/gallery/index.js]: I think we may be able to simplify the transform code to the following:
```
				transform: ( attributes ) => {
					// Init the align attribute from the first item which may be either the place holder or an image.
					let { align } = attributes[ 0 ];
					// Loop through all the images and check if they have the same align.
					align = every( attributes, [ 'align', align ] ) ? align : undefined;

					const validImages = filter( attributes, ( { id, url } ) => id && url );

					return createBlock( 'core/gallery', {
						images: validImages.map( ( { id, url, alt, caption } ) => ( { id, url, alt, caption } ) ),
						ids: validImages.map( ( { id } ) => id ),
						align,
					} );
				},
```.

With the added advantage that the loop to verify if the aligns are the same stops as soon as the first difference is found, while the reduce continue the iteration.

**Review by jorgefilipecosta — COMMENTED** [review]
Thank you for your contribution @andreilupu, everything tested well in my tests. I added a possible improvement that may simplify the images to gallery transform.

File: packages/block-library/src/gallery/index.js (hunk 1 above)
Comment (andreilupu) [reply packages/block-library/src/gallery/index.js]: That's a nice improvement, thanks! I've learned something today!

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍 Thank you for iterating on this changes!

**Comment by rowroll** [comment]
Thanks all, this will make my life easier! :)

**Pull request merged**